### PR TITLE
Feature/hdf5 stack reader

### PR DESCRIPTION
**Pull request by k-dominik**
This complements
* https://github.com/ilastik/lazyflow/pull/232 and
* https://github.com/ilastik/lazyflow/pull/226
and exposes the functionality in the ilastik-Gui

So far only multi-file h5-files can be imported via the usual way. I couldn't run the tests, so I'm not 100% sure that nothing is broken due to the added functionality. Will try it on a different machine later today.

### CODE REVIEW

**Comment by k-dominik** [comment]
So, now I can run the tests and none of them is failing because of the changes here. However, I'm in the process of writing a tests that checks the file loading in general

**Comment by k-dominik** [comment]
Together with the mentioned lazyflow PRs (https://github.com/ilastik/lazyflow/pull/232 and ilastik/lazyflow#226) this PR closes https://github.com/ilastik/ilastik/issues/1361

Note that the changes from https://github.com/ilastik/lazyflow/pull/232 need to be merged for this PR to work.

With the latest commits I have also generalized the tests for `OpDataSelection` to test for all vigra types and h5 (2D, 3D, 3D stacks).

File: ilastik/applets/dataSelection/dataSelectionSerializer.py [hunk 1]
```diff
@@ -212,6 +218,37 @@ def importStackAsLocalDataset(self, info, sequence_axis='t'):
...
+                    OpStreamingHdf5SequenceReaderM.InternalPlaceholderError):
+                isMultiFile = False
+
+            assert (not(isMultiFile and isSingleFile)), (
+                "Something is wrong, glob string shouldn't allow both")
+            assert (isMultiFile or isSingleFile), (
+                "Glob string doesn't conform to h5 stack glob string rules")
+
+            if isSingleFile:
+                opLoader = OpStreamingHdf5SequenceReaderS(parent=self.topLevelOperator.parent)
```
Comment (stuarteberg) [inline ilastik/applets/dataSelection/dataSelectionSerializer.py]: In this case, `opLoader.Hdf5File` is never configured, causing an error.

**Comment by stuarteberg** [comment]
In addition to the comment above, I ran into a couple of minor issues, which I've fixed in the following branch:
[ilastik/feature/hdf5StackReader](https://github.com/ilastik/ilastik/tree/feature/hdf5StackReader)

File: ilastik/applets/dataSelection/dataSelectionSerializer.py (hunk 1 above)
Comment (k-dominik) [inline ilastik/applets/dataSelection/dataSelectionSerializer.py]: I have addressed this issue by changing the interface of `OpStreamingHdf5SequenceReaderS` in lazyflow. The hdf5 file is now handled in this class and derived from the `GlobString`. The attribute `Hdf5File` was, therefore, removed.

**Comment by k-dominik** [comment]
Thank you for checking at the PR!
 
And also thank you for the commits to the [ilastik/ilastik/feature/hdf5StackReader](https://github.com/ilastik/ilastik/tree/feature/hdf5StackReader) branch.
I haven't merged your commit af2da20, see the comment there.

**Comment by k-dominik** [comment]
Okay, I merged your latest commit, should be good to go then. Still a bit unsatisfying, that i couldn't reproduce the behavior you were seeing.

**Pull request merged**